urx/urrobot.py

```diff
@@ -277,6 +277,17 @@ def movej(self, joints, acc=0.1, vel=0.05, wait=True, relative=False, threshold=
             self._wait_for_move(joints[:6], threshold=threshold, joints=True)
             return self.getj()
 
+    def movej_to_pose(self, tpose, acc=0.1, vel=0.05, wait=True, relative=False, threshold=None):
```

Comment:
what is that method supposed to do? it looks like the same as movj or is it?

Reply:
hi, the movej_to_pose method allows to set a pose instead of joints as target of the movej. I haven't found that variant in the api.

Reply:
sorry forgot that one. Bit does that make sense at all? is tpose is a list of joint values then it is the same as movej, if it is a pose, then you need the inverse kinematic to find joint values to send... I really do not get what it does...

Reply:
If i pass a pose variable to movej the robot itself calculates the kinematic. You can try this yourself within Polyscope. You can pass a pose to movej. IMHO this just makes it easier to use

Reply:
joint positions (**q can also be specified as a pose, then
inverse kinematics is used to calculate the corresponding
joint positions**)

From the script manual for movej

Reply:
did not know about it. Thanks

Reply:
maybe you should writ that comment in doc string so other people understand it?

Reply:
@Mofeywalker I am wondering is your ```movej_to_pose_list``` method working? BTW, I want to switch ON the welding torch and then use ```movej_to_pose_list```, however the ```set_digital_out``` method is sent over as a complete program. When it finishes there is no program running and hence the digital output will fall back to OFF for safety sake. Thank you for your help.

Reply:
> 
> 
> @Mofeywalker I am wondering is your `movej_to_pose_list` method working? BTW, I want to switch ON the welding torch and then use `movej_to_pose_list`, however the `set_digital_out` method is sent over as a complete program. When it finishes there is no program running and hence the digital output will fall back to OFF for safety sake. Thank you for your help.

hey @blackhorsewu the last time i worked on this in december 2018 it worked :) But I changed jobs in the meantime and don't have access to an UR anymore.

Reply:
@Mofeywalker thank you very much for your reply. I would then try on it. Thank you also very much for the movej_to_pose. That solved my problem. I was trying to find out ways to call get_inverse_kin() and was getting nowhere! Anyway thank you again for the good work.